internals/constants/index.ts

```diff
@@ -0,0 +1,12 @@
+export const PROJECT_ENV_TARGET = {
+  Production: 'production',
+  Preview: 'preview',
+  Development: 'development'
+} as const;
+
+export const PROJECT_ENV_TYPE = {
+  Plaintext: 'plain',
+  Secret: 'secret',
+  Encrypted: 'encrypted',
+  System: 'system'
+} as const;
```

Comment:
I wonder if we could eliminate enums and instead do something like

```ts
export const type ProjectEnvType = 'plain' | 'secret' | 'encrypted' | 'system'
```

Reply:
Nope because these strings are used as literals throughout the code base

Reply:
I don't understand. It also seems to me that we could use the string literals for this.

Reply:
I think this comes down to preference. If we only have the type, then throughout the app wherever you'd previously get to do `ProjectEnvType.Encrypted`, you now have to know to write `'encrypted'`. Luckily, type preview in VSCode does help with that. And typescript will enforce we are getting it write.

I for one enjoy the pattern of `as const` objects to emulate `enum`. I don't know where else we might have this pattern in the code base. IMO more of our _constant_ string values should be defined this way. Making updates is simpler because you just have to modify the singular string rather than everywhere it might be copied too (think tests, similar code, etc.).

Happy to bikeshed this some more. Lots of these refactor PRs are intended to expose some of the codebase's _smell_. More than happy to clean it up as I go instead of carrying it along lol

Reply:
> Happy to bikeshed this some more

Lets try the `export type ProjectEnvType = 'plain' | 'secret' | 'encrypted' | 'system'` solution and see if it can be refactored in 30 min or less (timebox it)

Reply:
And you can move it from `internal/constants` to `internal/types` 👍

Reply:
> you just have to modify the singular string rather than everywhere it might be copied too

In theory, yes, but in practice, I've always updated the property name as well because it would be confusing otherwise, which meant I was updated it everywhere regardless.

The string literal version is (in my opinion) easier to write, easier to read, and just as type safe.

I'm fine with anything that's not a typescript enum, though.

Reply:
Refactored to just use types.